src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -596,18 +808,144 @@ lookAhead (Parser step1 initial1 _) =
 -------------------------------------------------------------------------------
 -- Interleaving
 -------------------------------------------------------------------------------
+
+data ParserTurn = Parser1 | Parser2
+
 --
 -- | See 'Streamly.Internal.Data.Parser.deintercalate'.
 --
--- /Unimplemented/
+-- /Internal/
 --
 {-# INLINE deintercalate #-}
 deintercalate ::
-    -- Monad m =>
-       Fold m a y -> Parser m x a
-    -> Fold m b z -> Parser m x b
+    MonadCatch m
+    => Fold m a y 
+    -> Parser m x a
+    -> Fold m b z 
+    -> Parser m x b
     -> Parser m x (y, z)
-deintercalate = undefined
+deintercalate
```

Comment:
```
deintercalate fld1 par1 fld2 par2 = 
    many ((,) <*> lmap fst fld1 <*> lmap snd fld2) ((,) <*> par1 <*> par2)
```
There's probably a better way of writing the above.
Note that the behavior of  `<*>` is different for fold and parsers. This will change in the future.